kafka-client/src/main/java/dev/responsive/kafka/internal/db/partitioning/SegmentPartitioner.java

```diff
@@ -126,6 +129,8 @@ public SegmentPartitioner(
     this.retentionPeriodMs = retentionPeriodMs;
     this.segmentIntervalMs = segmentIntervalMs;
     this.windowSizeMs = windowSizeMs;
+    LOG.info("Created segment partitioner with retentionPeriod={}ms, segmentInterval={}ms,"
```

Comment:
can we assert that all three of these values are strictly greater than 0? that way we'll avoid all division by zero (we should update `StoreUtil#computeSegmentInterval` to make sure segmentInterval can't be 0)

Reply:
Sure thing. By the way, I lowered the default num_segments by a lot (a few weeks ago, not in this PR) so I _think_ even without this fix, you should be able to run the window QTT tests without hitting the div by zero issue. Of course it doesn't hurt to just wait for the actual fix either